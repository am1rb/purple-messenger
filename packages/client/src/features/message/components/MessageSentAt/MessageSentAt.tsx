import React, { useEffect, useState } from "react";
import moment from "moment";
import { Message } from "@purple-messenger/core";
import { Typography } from "@material-ui/core";

export type MessageSentAtProps = Pick<Message, "sentAt">;

function MessageSentAt({ sentAt }: MessageSentAtProps) {
  const [, forceUpdate] = useState(0);

  useEffect(() => {
    const timerRef = setInterval(() => forceUpdate((v) => v + 1), 60 * 1000);
    return () => clearInterval(timerRef);
  }, []);

  return (
    <Typography variant="caption" color="textSecondary">
      {moment(sentAt).fromNow()}
    </Typography>
  );
}

export default MessageSentAt;
