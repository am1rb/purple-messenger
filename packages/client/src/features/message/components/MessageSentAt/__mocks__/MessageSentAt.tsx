import React from "react";

const messageSentAt = jest.fn(() => <span data-testid="message-sent-at" />);

export default messageSentAt;
