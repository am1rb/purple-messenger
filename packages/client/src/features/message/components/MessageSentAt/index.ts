export { default } from "./MessageSentAt";
export * from "./MessageSentAt";
