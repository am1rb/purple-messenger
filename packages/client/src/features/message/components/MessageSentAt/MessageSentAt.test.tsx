import React from "react";
import { render } from "@testing-library/react";
import MessageSentAt from "./MessageSentAt";

describe("The <MessageSentAt /> tests", () => {
  let dateSpy: jest.SpyInstance;

  beforeAll(() => {
    dateSpy = jest
      .spyOn(global.Date, "now")
      .mockImplementation(() => Date.parse("2020-02-14"));
  });
  afterAll(() => dateSpy.mockRestore());

  it("Should match the snapshot", () => {
    const { container } = render(
      <MessageSentAt sentAt={new Date("2020-02-13")} />
    );
    expect(container).toMatchSnapshot();
  });
});
