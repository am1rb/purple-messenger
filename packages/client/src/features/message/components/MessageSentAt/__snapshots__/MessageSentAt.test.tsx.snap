// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <MessageSentAt /> tests Should match the snapshot 1`] = `
<div>
  <div
    data-testid="mui-typography"
  >
    a day ago
  </div>
</div>
`;
